lib/portscan_handler.py

```diff
@@ -0,0 +1,61 @@
+import uuid
+import logging
+import boto3
+import json
+import os
+
+from lib.target import Target
+from lib.response import Response
+from lib.hosts import Hosts
+
+
+class PortScanHandler(object):
+    def __init__(self, sqs_client=boto3.client('sqs', region_name='us-west-2'), logger=logging.getLogger(__name__), region='us-west-2'):
+        self.sqs_client = sqs_client
+        self.logger = logger
+        self.region = region
+
+    def queue(self, event, context):
+        data = json.loads(event['body'])
+        if "target" not in data:
+            self.logger.error("Unrecognized payload")
+            return Response({
+                "statusCode": 500,
+                "body": json.dumps({'error': 'Unrecognized payload'})
+            }).with_security_headers()
+
+        target = Target(data.get('target'))
+        if not target:
+            self.logger.error("Target validation failed of: " +
+                              target.name)
+            return Response({
+                "statusCode": 400,
+                "body": json.dumps({'error': 'Target was not valid or missing'})
+            }).with_security_headers()
+
+        scan_uuid = str(uuid.uuid4())
+        print(self.sqs_client.send_message(
```

Comment:
I know this is from the original code, but we don't need the `print` here to put a message into the queue.

Reply:
removed.